src/Microsoft.OpenApi/Extensions/OpenApiTypeMapper.cs

```diff
@@ -12,40 +12,81 @@ namespace Microsoft.OpenApi.Extensions
     /// </summary>
     public static class OpenApiTypeMapper
     {
+        /// <summary>
+        /// Maps a JsonSchema data type to an identifier.
+        /// </summary>
+        /// <param name="schemaType"></param>
+        /// <returns></returns>
+        public static string ToIdentifier(JsonSchemaType? schemaType)
+        {
+            return schemaType switch
+            {
+                JsonSchemaType.Null => "null",
+                JsonSchemaType.Boolean => "boolean",
+                JsonSchemaType.Integer => "integer",
+                JsonSchemaType.Number => "number",
+                JsonSchemaType.String => "string",
+                JsonSchemaType.Array => "array",
+                JsonSchemaType.Object => "object",
+                _ => null,
+            };
+        }
+
+        /// <summary>
+        /// Converts a schema type's identifier into the enum equivalent
+        /// </summary>
+        /// <param name="identifier"></param>
+        /// <returns></returns>
+        public static JsonSchemaType IdentifierToEnumType(string identifier)
+        {
+            return identifier switch
+            {
+                "null" => JsonSchemaType.Null,
+                "boolean" => JsonSchemaType.Boolean,
+                "integer" or "int" => JsonSchemaType.Integer,
+                "number" or "double" => JsonSchemaType.Number,
+                "string" => JsonSchemaType.String,
+                "array" => JsonSchemaType.Array,
+                "object" => JsonSchemaType.Object,
+                "file" => JsonSchemaType.String, // File is treated as string
+                _ => JsonSchemaType.Any,
```

Comment:
Remove JsonSchemaType.Any  and in this case throw an exception.  It should not stop parsing, but convert to a problem. I think this happens in ParseNode.